pubtools/pulplib/_impl/model/maintenance.py

```diff
@@ -0,0 +1,179 @@
+import logging
+import datetime
+import json
+import os
+
+import jsonschema
+
+from pubtools.pulplib._impl import compat_attr as attr
+from ..schema import load_schema
+from .common import InvalidDataException
+
+
+LOG = logging.getLogger("pubtools.pulplib")
+
+
+USER = os.environ.get("USER")
+HOSTNAME = os.environ.get("HOSTNAME")
+OWNER = "%s@%s" % (USER, HOSTNAME)
+
+
+def iso_time_now():
+    return datetime.datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+
+@attr.s(kw_only=True, frozen=True)
+class MaintenanceEntry(object):
+    """Details about the maintenance"""
+
+    id = attr.ib(default=None, type=str)
+    """ID of repository in maintenance"""
+    message = attr.ib(default=None, type=str)
+    """Why this repository is in maintenance"""
+    owner = attr.ib(default=None, type=str)
+    """Who set this repository in maintenance mode"""
+    started = attr.ib(default=None, type=datetime.datetime)
+    """:class:`~datetime.datetime` in UTC at when the maintenance started"""
+
+
+@attr.s(kw_only=True, frozen=True)
+class MaintenanceReport(object):
+    """Represents the maintenance status of Pulp repositories
```

Comment:
Current doc style is to use full sentences with punctuation, so:

```suggestion
    """Represents the maintenance status of Pulp repositories.
```